### PR TITLE
[OSD-6207] Add Misspell to AAO

### PR DESCRIPTION
**Pull request by dkeohane**
[WIP]
[OSD-6207](https://issues.redhat.com/browse/OSD-6207)

- Added Misspell to AAO to run against pkg and docs

### CODE REVIEW

**Comment by iamkirkbater** [comment]
These won't pass because you're modifying the `boilerplate` files, which have file integrity checks on them.  This would need to be committed to either Boilerplate itself upstream (openshift/boilerplate) or it would need to be added in here some other way.

**Comment by dkeohane** [comment]
> These won't pass because you're modifying the `boilerplate` files, which have file integrity checks on them. This would need to be committed to either Boilerplate itself upstream (openshift/boilerplate) or it would need to be added in here some other way.

Yeah I'd chatted with @drpaneas about this already - holding off on updating this til he gets his answers on GH actions - thanks for being responsive :)

**Review by drpaneas — COMMENTED** [review]
Also, I guess we have to update the documentation saying that the best way to run this is: `./boilerplate/_lib/container-make check-spell`

**Comment by dastergon** [comment]
@dkeohane for full context, could you update the PR with a short description and what it solves?

File: Makefile [hunk 1]
```diff
@@ -328,7 +328,18 @@ delete-sts-accountclaim: ## Deletes a templated STS accountclaim
...
+# GOLANGCI_LINT_CACHE needs to be set to a directory which is writeable
+# Relevant issue - https://github.com/golangci/golangci-lint/issues/734
+GOLANGCI_LINT_CACHE ?= /tmp/golangci-cache
+
+# Spell Check
+.PHONY: check-spell
+check-spell: # Check spelling
+	GOLANGCI_LINT_CACHE=${GOLANGCI_LINT_CACHE} golangci-lint run -c config/golangci-lint-config.yml
+
+lint: check-spell
```
Comment (2uasimojo) [inline Makefile:336]: I would include a comment here, something like:
```suggestion
# This *adds* `check-spell` to the existing `lint` provided by boilerplate
lint: check-spell
```

**Comment by dastergon** [comment]
Is that still a WIP? if not, we can remove the WIP tags from the PR

**Comment by drpaneas** [comment]
Now it's ready for review

File: docs/1.0-Installation.md [hunk 2]
```diff
@@ -12,14 +12,11 @@ Quick Access:
     - [1.5.2 Cluster Mode](#152-cluster-mode)
 
 
-
 ## 1.1 - Prerequisites
 Please ensure you have completed all necessary prerequisite steps before moving onto the Workflow Section below.
 
 [All required prerequisites steps can be found in this section.](1.1-InstallationPrerequisites.md)
 
-Please utilise the [Glossary](0.1-Glossary.md) for any terms or acronyms found in this documentation which you may find unclear.
```
Comment (dkeohane) [inline docs/1.0-Installation.md:21]: Is there a reason to remove this?

**Review by dkeohane — COMMENTED** [review]
Changes look good @drpaneas :taco:

File: docs/1.0-Installation.md (hunk 2 above)
Comment (drpaneas) [reply docs/1.0-Installation.md:21]: Hm? Should slipped during rebase. Feel free to add it back.

**Pull request merged**